stackslib/src/chainstate/stacks/boot/signers-voting.clar

```diff
@@ -39,21 +54,25 @@
 (define-read-only (get-vote (reward-cycle uint) (round uint) (signer principal))
     (map-get? votes {reward-cycle: reward-cycle, round: round, signer: signer}))
 
+(define-read-only (get-candidate-info (reward-cycle uint) (round uint) (candidate (buff 33)))
+    {candidate-weight: (default-to u0 (map-get? tally {reward-cycle: reward-cycle, round: round, aggregate-public-key: candidate})),
+    total-weight: (map-get? cycle-total-weight reward-cycle)})
+
 (define-read-only (get-tally (reward-cycle uint) (round uint) (aggregate-public-key (buff 33)))
     (map-get? tally {reward-cycle: reward-cycle, round: round, aggregate-public-key: aggregate-public-key}))
 
-(define-read-only (get-current-signer-slots (signer-index uint))
+(define-read-only (get-current-signer-weight (signer-index uint))
     (let ((cycle (+ u1 (burn-height-to-reward-cycle burn-block-height))))
-      (get-signer-slots signer-index cycle)))
+      (get-signer-weight signer-index cycle)))
```

Comment:
I'd suggest deleting `get-current-signer-weight`. Any caller can just use `get-signer-weight` and handle figuring out the current reward cycle themselves. I think the invocation in `vote-for-aggregate-public-key` is potentially dangerous as well: that function takes in a `reward-cycle` argument, but invokes `get-current-signer-weight`, so I think there could end up being a mismatch in the queried reward cycle.

Reply:
Yes, good catch! Fixed in f596c0bf5